src/helpers/Sentry.ts

```diff
@@ -1,26 +1,48 @@
 import * as Sentry from '@sentry/browser';
 import * as Integrations from '@sentry/integrations';
 import createSentryMiddleware from 'redux-sentry-middleware';
+import { getType } from 'typesafe-actions';
+import { recordSentryMessage } from '../app/errors/actions';
+import { AnyAction, Middleware, } from '../app/types';
 import config from '../config';
 
 let IS_INITIALIZED = false;
 
+const ignoredActions: Set<AnyAction['type']> = new Set([
+  recordSentryMessage,
+].map(getType));
+
+const filterBreadcrumbActions = (action: AnyAction) => !ignoredActions.has(action.type);
+
 export default {
 
-  initializeWithMiddleware() {
-    Sentry.init({
-      dist: config.RELEASE_ID,
-      dsn: 'https://84d2036467d546038347f0ac9ccd8b3b:c815982d89764df583493a60794e54aa@sentry.cnx.org/17',
-      environment: config.DEPLOYED_ENV,
-      integrations: [
-        new Integrations.ExtraErrorData(),
-        new Integrations.CaptureConsole(),
-        new Integrations.Dedupe(),
-      ],
-      release: `rex@${config.RELEASE_ID}`,
-    });
-    IS_INITIALIZED = true;
-    return createSentryMiddleware(Sentry);
+  initializeWithMiddleware(): Middleware {
+    return (store) => {
+      Sentry.init({
+        beforeSend: (event: Sentry.Event) => {
```

Comment:
i see you filtering out the `recordSentryMessage` actions from the breadcrumbs sent to sentry, is that because this `beforeSend` is called with each breadcrumb? i as thinking it would only be called with errors, if its also called with breadbrumbs we might want to do some more filtering so only error event ids are saved

Reply:
Well every action gets added as a breadcrumb in the [middleware](https://github.com/vidit-sh/redux-sentry-middleware/blob/4d05e62f843441aa5015221bb0cadb399c8f3894/src/main.js#L45) and based on `Sentry supports a concept called Breadcrumbs, which is a trail of events which happened prior to an issue.` (from their documentation) i assumed that it clears them after it saves (?) an issue.

Reply:
I'm not sure how the actual error raport looks, but isnt it fine if there is an error with breadcrumb of actions that lead to it? We could add `&& event.level === 'error'` to the if in `beforeSend` so when users screenshot the error modal they will show the error id only which will have all the breadcrumbs saved wherever sentry saves them

Reply:
i want all sentry events to be captured by this logic, but breadcrumbs are not events. a breadcrumb is created for every redux action that is dispatched, and that will be way too much noise for the ids being shown in the error dialog

Reply:
I think it works better now, but I'm not sure if actions should be included in breadcrumbs, and if yes then which ones.

Reply:
i'm confused, i don't think we have to filter breadcrumbs at all. the existing behavior has breadcrumbs for all redux actions, and we want to keep that.

but its important that we only store the ids for actual error events. for one because storing the ids for breadcrumbs would be super noisy, but also because the action to store the id would create an infinite loop.

Reply:
i just did some testing and it seems like the beforeSend function is not called with breadcrumbs, its only called with errors, so i don't think any filtering is necessary

Reply:
And because of that im pretty sure only error ids are being saved